src/FqnUtilities.php

```diff
@@ -22,7 +22,7 @@ function getFqnsFromType($type): array
     }
     if ($type instanceof Types\Compound) {
         for ($i = 0; $t = $type->get($i); $i++) {
```

Comment:
Wouldn't it be better to avoid assignment in the `foreach` statement and use `do .. while` instead? That would make the code more readable.